File: src/main/java/com/rg/webmvctest/WebMvcTestApplication.java
package com.rg.webmvctest;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication
public class WebMvcTestApplication {

    public static void main(String[] args) {
        SpringApplication.run(WebMvcTestApplication.class, args);
    }

    /**
     * For Spring Boot 1.4.x post-processor is required to perform validation on request body using: javax.validation.Valid.
     *
     * We don't use it here since it is a Spring Boot 1.5.x based app.
     */
//    @Bean
//    public MethodValidationPostProcessor methodValidationPostProcessor() {
//        return new MethodValidationPostProcessor();
//    }
}

File: src/main/kotlin/com/rg/webmvctest/WebMvcTestApplication.kt
package com.rg.webmvctest

import com.fasterxml.jackson.module.kotlin.KotlinModule
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean
import org.springframework.validation.beanvalidation.MethodValidationPostProcessor

@SpringBootApplication
open class WebMvcTestApplication {

    /**
     * This post-processor is required to perform validation on request body using: javax.validation.Valid
     */
    @Bean
    open fun methodValidationPostProcessor(): MethodValidationPostProcessor = MethodValidationPostProcessor()

    @Bean
    open fun kotlinModule(): KotlinModule = KotlinModule()
}

fun main(args: Array<String>) {
    SpringApplication.run(WebMvcTestApplication::class.java, *args)
}
